docs/quickstarts/tanstack-start.mdx

```diff
@@ -0,0 +1,246 @@
+  - Add `<ClerkProvider/>` to your application
+  - Protecting your pages
+</TutorialHero>
+
+Learn how to use Clerk to quickly and easily add secure authentication and user management to your TanStack Start application.
+
+<Steps>
+  ### Install `@clerk/tanstack-start`
+
+  Once you have a TanStack Start application ready, you need to install Clerk's TanStack Start SDK. This gives you access to our prebuilt components and hooks for TanStack Start applications.
+
+  <CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
+    ```bash {{ prettier: false, filename: 'terminal' }}
+    npm install @clerk/tanstack-start
+    ```
+
+    ```bash {{ prettier: false, filename: 'terminal' }}
+    yarn add @clerk/tanstack-start
+    ```
+
+    ```bash {{ prettier: false, filename: 'terminal' }}
+    pnpm add @clerk/tanstack-start
+    ```
+  </CodeBlockTabs>
+
+  ### Set environment variables
+
+  Below is an example of an `.env` file.
+
+  **Pro tip!** If you are signed into your Clerk Dashboard, your secret key should become visible by clicking on the eye icon. Otherwise, you can find your keys in the Clerk Dashboard on the [API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page.
+
+  ```env {{ prettier: false, filename: '.env' }}
+  VITE_CLERK_PUBLISHABLE_KEY={{pub_key}}
+  CLERK_SECRET_KEY={{secret}}
+  ```
+
+  ### Add `clerkHandler` to your application
+
+  To configure Clerk in your TanStack application, you will need to update SSR entrypoint.
+
+  ```tsx {{ prettier: false, filename: 'app/ssr.tsx' }}
+  import { createStartHandler, defaultStreamHandler } from '@tanstack/start/server';
+  import { getRouterManifest } from '@tanstack/start/router-manifest';
+  import { createRouter } from './router';
+  import { createClerkHandler } from '@clerk/tanstack-start/server';
+
+  export default createClerkHandler(createStartHandler({
+    createRouter,
+    getRouterManifest,
+  }))(defaultStreamHandler);
+  ```
+
+  ### Add `<ClerkProvider>` and components to your app
+
+  All Clerk hooks and components must be children of the [`<ClerkProvider>`](/docs/components/clerk-provider) component, which provides active session and user context.
+
+  You can control which content signed in and signed out users can see with Clerk's [prebuilt components](/docs/components/overview). To get started, create a header for your users to sign in or out. To do this, you will use:
+
+  - [`<SignedIn>`](/docs/components/control/signed-in): Children of this component can only be seen while **signed in**.
+  - [`<SignedOut>`](/docs/components/control/signed-out): Children of this component can only be seen while **signed out**.
+  - [`<UserButton />`](/docs/components/user/user-button): A prebuilt component that comes styled out of the box to show the avatar from the account the user is signed in with.
+  - [`<SignInButton />`](/docs/components/unstyled/sign-in-button): An unstyled component that links to the sign-in page or displays the sign-in modal.
```

Comment:
The example below doesn't do that, it only adds the Provider 😊 So we either adjust the example or this text